src/solvers/kriging.jl

```diff
@@ -180,36 +177,24 @@ function solve_approx(problem::EstimationProblem, var::Symbol, preproc)
     neighbors = Vector{Int}(undef, maxneighbors)
     X = Matrix{coordtype(pdomain)}(undef, ndims(pdomain), maxneighbors)
 
-    # keep track of estimated locations
-    estimated = falses(npoints(pdomain))
-    for (loc, datloc) in datamap(problem, var)
-      varμ[loc] = pdata[datloc,var]
-      varσ[loc] = zero(V)
-      estimated[loc] = true
-    end
-
     # estimation loop
     for location in traverse(pdomain, path)
-      if !estimated[location]
-        # coordinates of neighborhood center
-        coordinates!(xₒ, pdomain, location)
-
-        # find neighbors with previously estimated values
-        nneigh = search!(neighbors, xₒ, bsearcher, mask=estimated)
-
-        # skip location in there are too few neighbors
-        if nneigh < minneighbors
-          varμ[location] = NaN
-          varσ[location] = NaN
-          estimated[location] = true
-          continue
-        end
+      # coordinates of neighborhood center
+      coordinates!(xₒ, pdomain, location)
+
+      # find neighbors with previously estimated values
+      nneigh = search!(neighbors, xₒ, bsearcher)
 
+      # skip location in there are too few neighbors
+      if nneigh < minneighbors
+        varμ[location] = NaN
+        varσ[location] = NaN
```

Comment:
Just a comment that in a future release we should perhaps consider only `missing` as the sentinel. The `NaN` here is a temporary hack I've adopted, but certainly deserves a future review :+1: